### DISCUSSION TITLE
Doubts regarding pop up window of the tiddlyclip add on

### DISCUSSION THREAD

**Discussion by arunnbabu81** [buggyj/tiddlyclip-plugin#49] [Q&A] [answered]
<img width="608" alt="image" src="https://user-images.githubusercontent.com/67494083/149401285-b8224624-ed3e-4d3a-94f3-6ad2551606ba.png">

1. How to make use of the text box area in the upper part of this pop up window
2. Is it possible to snip text which are annotated using the colour selection in the lower part of this pop up window or is it just for taking snap

**Comment by buggyj** [comment, marked as answer]
# Answer part 1

If 'note' is added to the modes column in the TiddlyClipConfig then the content of the 'note' box is sent to the plugin in the variable '@note', eg
![image](https://user-images.githubusercontent.com/2855795/149549768-c8326a85-213e-46de-87ef-6b9003edb744.png)

![image](https://user-images.githubusercontent.com/2855795/149549892-1bf30418-ee6a-468a-9a84-ab9e5247c735.png)

Text can be written directly into the 'note' box or clipped from a page using 'text2note' like so:

![image](https://user-images.githubusercontent.com/2855795/149550854-bfabde8b-fb12-4ab0-9942-aa7555907c29.png)

----

# Answer part 2

Highlights applied using the 4 colors in the popup will appear in the '@web' (clipped html) as well as snaps.

**Reply by arunnbabu81** [reply]
1. text2note was working. When I select text/multiple text from a webpage and click text2note, it gets added to the "note" box. 
But how to send these text added to the "note" box to tiddlywiki?

2. What is WordList supposed to do? When I click on WordList from the Firefox context menu, a notification appears in the tiddlywiki tab like this 
<img width="725" alt="image" src="https://user-images.githubusercontent.com/67494083/149660716-3abd3c55-c601-49d0-b9e6-c01c0dbb0334.png">

And if I click on ok , this error message comes up

<img width="639" alt="image" src="https://user-images.githubusercontent.com/67494083/149660761-ec76f269-d0aa-4c82-99c8-35d71d8b5039.png">

**Reply by arunnbabu81** [reply]
If you don't mind, what does the PronounceList, Transcript, stream, streamclip, LogBook, highlight, select categories shown in the screenshot of the TiddlyClipConfig  do?

**Reply by buggyj** [reply]
The contains of wordlist should be:

`\rules only table wikilink prettylink`

`|!Title|!Body|!Tags|!Fields modifiers|!field initials|!Modes|`
`|WordList|((*@note*))\n\n||||append add|`

**Reply by buggyj** [reply]
the first line is optional - it just makes the tiddler easier to read.

**Reply by buggyj** [reply]
So in tiddlyclip [[WordList]] is an instruction to load a rule, as in discussion 
'How to apply a regular expression to use only a part of a URL as a title.' #47

**Reply by buggyj** [reply]
Usually, in the title column of a TiddlyClipConfig, we put the name of the tiddler that will contain the clip. Also a simple substitution eg '((@url))' can be placed directly in the title column. However if the title is to be the result of some rule,  then the rule, within double square brackets, in put in the title column.

**Reply by buggyj** [reply]
'highlight' will add a background color to selected text.

Sometime it is difficult to select a paragraph of text. If a word is selected with the mouse and then 'select' is actioned then the amount of selected text will expand to a paragraph (sometime it is necessary to apply this more than one) .

**Reply by buggyj** [reply]
The AmiAireRule is
![image](https://user-images.githubusercontent.com/2855795/149665098-9380d5e0-c10d-4946-841d-9c655f1797db.png)

It is used for podcast transcirpts. Here I change the created tiddler type so that I can add hover-over definitions/translations to spanish words that I don't know, using the ckeditor.

**Reply by arunnbabu81** [reply]
- highlight and select are useful features, especially select.

- I didn't fully understood how to send the text within the note box into tw. 

     So I made a new rule called snipnote 

<img width="489" alt="image" src="https://user-images.githubusercontent.com/67494083/149676241-6fe0c855-d93c-42ed-bc41-55fd974d75a4.png">

<img width="933" alt="image" src="https://user-images.githubusercontent.com/67494083/149676258-fad3c67b-50cb-4cff-b366-7a85dd3901e9.png">

This is working for the timebeing.
How am I actually supposed to send text from the note box into tw?

**Reply by buggyj** [reply]
That is correct, and what I was attempting to show in the first part of my reply (the first line of my response)

**Reply by arunnbabu81** [reply]
> The AmiAireRule is ![image](https://user-images.githubusercontent.com/2855795/149665098-9380d5e0-c10d-4946-841d-9c655f1797db.png)
> 
> It is used for podcast transcirpts. Here I change the created tiddler type so that I can add hover-over definitions/translations to spanish words that I don't know, using the ckeditor.

Is visual editor available as a packaged plug in? will it work with TW 5.1.23 or higher?

**Reply by buggyj** [reply]
Yes it works with all versions of TW as far as I know. It needs this version of ckeditor lib to be download on to your computer: https://ckeditor.com/cke4/release/CKEditor-4.3.2

The instructions to install are here:

http://bjtools.tiddlyspot.com/#VisualEditor

**Reply by arunnbabu81** [reply]
I got the visual editor installed in my wiki ? Can you tell briefly in what all circumstances visual editor is useful.

Will this work with the new ckeditor5. The floating toolbar in ckeditor 5 looks interesting

**Reply by buggyj** [reply]
It does not work with all later versions, I have not tried with version 5.

There are a number of uses for visual editing, sometime I clip html that needs to be 'corrected' to display properly. Sometimes I want to add some annotations to the clipped html, and with ckeditor it is much easier to read what you are editing than to read the html. Ckeditor also has a large number of plugins that can be added.
